blogs/docker-ci-tricks-2/blog.md

```diff
@@ -179,14 +215,21 @@ WORKDIR /home/ci-user/git-machete
 
 ```
 
-The trick is to create a new user with the same user and group ID as your user on the host.
+In order to be able to build docker for local runs as non-root user and for the CI/CD runs on CircleCi's host machines as root user,
+we used multi-stage builds. Single dockerfile stage is defined with `FROM` <base_stage_being_build_upon> `AS` <name_of_the_new_stage> and executes each statement
+until the next `FROM` - defining a new stage. 
+
+We can select which stage to build using docker-compose's `target` argument. \
+e.g. when the `TARGET=LOCAL` and `MOUNT_POINT=home/ci-user`, 
+the `CIRCLE_CI` stage is skipped and `LOCAL` stage gets build (and each stage it builds upon, which in this case is `base`)
+
+### Run locally as a non-root user
+The trick is to create a new user with the same user and group ID as your user on the local machine.
 This is accomplished by
 `docker-compose build --build-arg user_id="$(id -u)" --build-arg group_id="$(id -g)" tox`
 in `ci/tox/local-run.sh`.
-If you use any modern Linux distribution and have just one non-root user on your machine, they're both likely equal to 1000 (see `UID_MIN` and `GID_MIN` in `/etc/login.defs`);
-on CircleCI VMs they both turn out to be 2000.
 
-We switch from `root` to the newly-created user by calling `USER ci-user`.
+We switch from docker's default `root` to the newly-created user by calling `USER ci-user`.
```

Comment:
To be checked where this default actually comes from &mdash; I'm not sure myself, it might not be imposed by Docker but just happen to be defined by the `alpine` base image

Reply:
Pls do a quick research &mdash; although there might not even be a clear answer out there 😅 in which case just
```suggestion
We switch from default `root` to the newly-created user by calling `USER ci-user`.
```
is okay